File: app/src/main/java/com/binaryninja/readerhub/wellcomewizard/WizardMainViewModel.java
package com.binaryninja.readerhub.wellcomewizard;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

public class WizardMainViewModel {
    private static MutableLiveData<String> setpager;
    private static MutableLiveData<String> fragment;
    public static void setPagerPos(String fragment){
        if (setpager==null)
            setpager = new MutableLiveData<>();
        setpager.setValue(fragment);
    }
    public static LiveData<String> getPagerPos(){
        if (setpager==null){
            setpager = new MutableLiveData<>();
            setpager.setValue("");
        }
        return setpager;
    }
    public static void changeFragment(String fragmentName){
        if (fragment==null)
            fragment = new MutableLiveData<>();
        fragment.setValue(fragmentName);
        setPagerPos("");
    }
    public static LiveData<String> handleChangeFragment(){
        if (fragment==null){
            fragment = new MutableLiveData<>();
            fragment.setValue("");
        }
        return fragment;
    }
}

File: app/src/main/java/com/binaryninja/readerhub/wellcomewizard/WizardMainViewModel.kt
package com.binaryninja.readerhub.wellcomewizard

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData

object WizardMainViewModel {
    private var setpager: MutableLiveData<String>? = null
    private var fragment: MutableLiveData<String>? = null
    fun setPagerPos(fragment: String) {
        if (setpager == null) setpager = MutableLiveData()
        setpager!!.value = fragment
    }

    val pagerPos: LiveData<String>?
        get() {
            if (setpager == null) {
                setpager = MutableLiveData()
                setpager!!.value = ""
            }
            return setpager
        }

    fun changeFragment(fragmentName: String) {
        if (fragment == null) fragment = MutableLiveData()
        fragment!!.value = fragmentName
        setPagerPos("")
    }

    fun handleChangeFragment(): LiveData<String>? {
        if (fragment == null) {
            fragment = MutableLiveData()
            fragment!!.value = ""
        }
        return fragment
    }
}